### PR TITLE
Add utility for opening the config file

### PR DESCRIPTION
**Pull request by neithernut**
This basically tries to load the config file.
One bit is still missing (waiting for #151 -- the JSON parser).
Targets #220.

### CODE REVIEW

File: src/util/config_file.c [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+ */
+
+int
+ws_config_load(void) {
+    int fd = open_config();
+    if (fd < 0) {
+        return fd;
+    }
+
+    //!< @todo fill in 2nd parameter
```
Comment (matthiasbeyer) [inline src/util/config_file.c]: What 2nd parameter?

File: src/util/config_file.c [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+    struct ws_deserializer* deserializer = NULL;
+    if (deserializer) {
+        return -1;
+    }
+
+    return ws_command_processor_new(fd, deserializer, NULL)? 0 : -1;
```
Comment (matthiasbeyer) [inline src/util/config_file.c]: Please add a space right before the `?` operator.

File: src/util/config_file.c [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+static int
+open_config(void) {
+    char* dir;
+    int res;
+
+    // check subdirectories of the home directory
+    {
+        char const* home = getenv(HOME);
+        size_t homelen = strlen(home);
+        char buf[homelen + 10];
```
Comment (matthiasbeyer) [inline src/util/config_file.c]: This `10` here appears to be a magic constant to me, so please document it.

File: src/util/config_file.c [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+        if (res >= 0) {
+            return res;
+        }
+    }
+
+    // check the xdg config directories
+    {
+        // setup all the buffers and temporaries
+        char const* xdg_dirs = getenv(XDG_CONFIG_DIRS);
+        char buf[strlen(xdg_dirs)+1];
```
Comment (matthiasbeyer) [inline src/util/config_file.c:189]: please add spaces around the `+` operator.

**Comment by matthiasbeyer** [comment]
Looks good besides the three remaining comments.

**Comment by neithernut** [comment]
I pushed some changes. The code should now be complete. But merging it having an _totally_ incomplete deserializer.

**Comment by neithernut** [comment]
Needs a little bit more work.

---

Disregard that. It's ready

**Comment by neithernut** [comment]
Testing. Alread found a few errors in the program flow (and fixed them).
I think the config loading function is ready now, but it seems the pipeline is still not working. I'll notify you as soon as I tracked down the problems.

**Comment by matthiasbeyer** [comment]
ACK.

**Comment by neithernut** [comment]
I encountered a bug in `ws_deserialize()` (it read an entire message but didn't return it). Investigating further. @matthiasbeyer, could you have a look at it? I put the following content in a file `waysome.json` in `$PWD`:

```
{ "UID": "my awesome uid", "commands": [ { "add": [1, 2] } ] }


```

**Comment by neithernut** [comment]
Turned out the user documentation was lying about the format.

**Comment by neithernut** [comment]
There were a few quite embarrasing bugs, but it looks like it all works now.

**Comment by matthiasbeyer** [comment]
Looks good to me. ACK.

**Pull request merged**